pwnlib/gdb.py

```diff
@@ -529,12 +614,21 @@ def debug(args, gdbscript=None, exe=None, ssh=None, env=None, sysroot=None, api=
     if env:
         env = {bytes(k): bytes(v) for k, v in env}
 
+    exe = which(packing._decode(exe or args[0]))
```

Comment:
Good catch, thanks!

So the check in line 655 calls `os.path.exists` which doesn't make sense in the SSH-Environment. But I'm unsure why `os.path.exists` is called in the first place, since thats already the job of the `which` call.

I checked the [commit](https://github.com/Gallopsled/pwntools/commit/f727b66475dc8eb779e2d0af56250892f7642fa9) + the related Pull Request:  #1323 but didn't find any explanation for the code.

Therefore I'm just going to delete the duplicate check.

Reply:
`os.path.exists` is there to check if the path passed in the `exe` parameter exists. Yes it's redundant when `exe` is None and it's grabbing `args[0]` using which. Now you're calling `which` on the `exe` parameter too.